Tests/TuistLoaderTests/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilderTests.swift

```diff
@@ -0,0 +1,71 @@
+import Path
+import XCTest
+@testable import TuistLoader
+@testable import TuistLoaderTesting
+@testable import TuistSupportTesting
+
+final class ProjectDescriptionHelpersBuilderTests: TuistUnitTestCase {
+    var projectDescriptionHelpersHasher: MockProjectDescriptionHelpersHasher!
+    var resourceLocator: ResourceLocator!
+    var helpersDirectoryLocator: MockHelpersDirectoryLocator!
+    var subject: ProjectDescriptionHelpersBuilder!
+
+    override func setUpWithError() throws {
+        super.setUp()
+        projectDescriptionHelpersHasher = MockProjectDescriptionHelpersHasher()
+        helpersDirectoryLocator = MockHelpersDirectoryLocator()
+        resourceLocator = ResourceLocator()
+
+        let cachePath: AbsolutePath = try temporaryPath()
+        let fileHandler = MockFileHandler(temporaryDirectory: { try self.temporaryPath() })
+        subject = ProjectDescriptionHelpersBuilder(
+            projectDescriptionHelpersHasher: projectDescriptionHelpersHasher,
+            cacheDirectory: cachePath,
+            helpersDirectoryLocator: helpersDirectoryLocator,
+            fileHandler: fileHandler
+        )
+    }
+
+    override func tearDown() {
+        projectDescriptionHelpersHasher = nil
+        helpersDirectoryLocator = nil
+        resourceLocator = nil
+        subject = nil
+        super.tearDown()
+    }
+
+    func test_build_dylid_once_for_unique_path_when_built_many_times() async throws {
+        // Given
+        let paths: [AbsolutePath] = [
+            "/path/to/helpers/1",
+            "/path/to/helpers/2",
+            "/path/to/helpers/3",
+        ].flatMap { path in
+            Array(repeating: path, count: 5)
+        }
+        .shuffled()
+
+        let projectDescriptionPath = try resourceLocator.projectDescription()
+        let searchPaths = ProjectDescriptionSearchPaths.paths(for: projectDescriptionPath)
+
+        system.defaultCaptureStubs = (nil, nil, 0)
```

Comment:
Is there a reason we can't stub the individual commands and instead we're allow some commands not to be stubbed?

Reply:
The command you need to stub looks veeeery complex. 
I went down this path at first, but then I thought that this test begins to check not only the behavior I need, but also the correctness of the command arguments, which makes it more fragile to refactoring.